### DISCUSSION TITLE
Authorization only via JWT (without UAA server)

### DISCUSSION THREAD

**Discussion by h0jeZvgoxFepBQ2C** [rabbitmq/rabbitmq-server#6093] [Questions]
Hi,

I just wanted to ask if there is a possibility to to authenticate clients with JWTs, but without a seperate UAA server?
(or I miss something and my question doesnt make any sense)?

F.e. I would like to set up a cert on the rabbitmq server where I can sign JWTs which clients can use then to connect via WebStomp - but only to the channels / actions which are included in the signed JWTs?

For me it's not really clear to me on how to do this with the current documentation, because it always refers to UAA servers?

Thank you!

**Comment by michaelklishin** [comment]
As of 3.11 there is support for [OIDC and more identity providers](https://blog.rabbitmq.com/posts/2022/07/oidc-integration/). "Self-signed JWTs" are not really a scenario we try to cover.

You can use [x.509 certificate-based authentication](https://github.com/rabbitmq/rabbitmq-server/tree/main/deps/rabbitmq_auth_mechanism_ssl) for clients with some protocols..

**Reply by h0jeZvgoxFepBQ2C** [reply]
Hm I see, is there any reason on why not to make self signed JWT a possible integration? Would make it a lot easier, since you only have to provide a certificate for the sign process and nothing more?

**Reply by michaelklishin** [reply]
We don't see enough demand for this case in the community, in particular amongst the largest users.

**Reply by h0jeZvgoxFepBQ2C** [reply]
I see, sad. So there is no easy way to do this as it seems :(

**Reply by h0jeZvgoxFepBQ2C** [reply]
Using x509 certificates is quite cumbersome, since I would have to generate each time new certificates for each user, I guess? Looks much easier with JTWs?

**Reply by michaelklishin** [reply]
If you ask me, using x.509 certificates is easier than setting up all the infrastructure for JWTs.

**Reply by michaelklishin** [reply]
@h0jeZvgoxFepBQ2C we don't have plans to support "self signed JWTs" at this time. RabbitMQ is open source software, you can contribute support for something like that if you really need this.
